solr/contrib/gcs-repository/src/java/org/apache/solr/gcs/GCSBackupRepository.java

```diff
@@ -0,0 +1,466 @@
+            throw new IllegalArgumentException("Error on creating URI", e);
+        }
+
+        return result;
+    }
+
+    @Override
+    public URI resolve(URI baseUri, String... pathComponents) {
+        StringBuilder builder = new StringBuilder(baseUri.toString());
+        for (String path : pathComponents) {
+            if (path != null && !path.isEmpty()) {
+                if (builder.charAt(builder.length()-1) != '/') {
+                    builder.append('/');
+                }
+                builder.append(path);
+            }
+        }
+
+        return URI.create(builder.toString());
+    }
+
+    @Override
+    public boolean exists(URI path) throws IOException {
+        if (path.toString().equals(getConfigProperty("location"))) {
+            return true;
+        }
+
+        if (path.toString().endsWith("/")) {
+            return storage.get(bucketName, path.toString(), Storage.BlobGetOption.fields()) != null;
+        } else {
+            final String filePath = path.toString();
+            final String directoryPath = path.toString() + "/";
+            return storage.get(bucketName, filePath, Storage.BlobGetOption.fields()) != null ||
+                    storage.get(bucketName, directoryPath, Storage.BlobGetOption.fields()) != null;
+        }
+
+    }
+
+    @Override
+    public PathType getPathType(URI path) throws IOException {
+        if (path.toString().endsWith("/"))
+            return PathType.DIRECTORY;
+
+        Blob blob = storage.get(bucketName, path.toString()+"/", Storage.BlobGetOption.fields());
+        if (blob != null)
+            return PathType.DIRECTORY;
+
+        return PathType.FILE;
+    }
+
+    private String toBlobName(URI path) {
+        return path.toString();
+    }
+
+    @Override
+    public String[] listAll(URI path) throws IOException {
+        String blobName = path.toString();
+        if (!blobName.endsWith("/"))
+            blobName += "/";
+
+        final String pathStr = blobName;
+        final LinkedList<String> result = new LinkedList<>();
+        storage.list(
+                bucketName,
+                Storage.BlobListOption.currentDirectory(),
+                Storage.BlobListOption.prefix(pathStr),
+                Storage.BlobListOption.fields())
+        .iterateAll().forEach(
+                blob -> {
+                    assert blob.getName().startsWith(pathStr);
+                    final String suffixName = blob.getName().substring(pathStr.length());
+                    if (!suffixName.isEmpty()) {
+                        // Remove trailing '/' if present
+                        if (suffixName.endsWith("/")) {
+                            result.add(suffixName.substring(0, suffixName.length() - 1));
+                        } else {
+                            result.add(suffixName);
+                        }
+                    }
+                });
+
+        return result.toArray(new String[0]);
```

Comment:
Now with Java 11 I prefer `result.toArray(String[]::new)`

Reply:
Gonna respectfully pass on this one - maybe when my work has me more in Java 11 codebases I'll prefer this too, but to my uncultured Java 8 eyes `String[]::new` makes me stop and think too long haha.  And it's one more thing to fix if I try an 8.x backport.